### DISCUSSION TITLE
Overflow Exception when AssertEquals_Bool or AssertEquals_STRING

### DISCUSSION THREAD

**Discussion by andimin182** [tcunit/TcUnit#230] [General]
Hey there,
I'm having quiet a lot of problem with a recurrent Overflow Exception when running the tests in TcUnit. I debugged the problem and i identify that this exception is thrown by the function: 
AssertMessageFormatter.LogAssertFailure(Expected := BOOL_TO_STRING(Expected),
                                            Actual := BOOL_TO_STRING(Actual),
                                            Message := Message,
                                            TestInstancePath := TestInstancePath);

In fact, in my case, there is a test that failes, hence it logs the failure but it gives me an overflow exception instead of correctly log the failure. I couldn't identify which argument is in overflow and why...
A second case, when instead the result == expected, Overflow exception is raised in another function that is:
AssertResults.ReportResult(ExpectedSize := SIZEOF(Expected),
                           ExpectedTypeClass := IBaseLibrary.TypeClass.TYPE_BOOL,
                           ExpectedValue := ADR(Expected),
                           ActualSize := SIZEOF(Actual),
                           ActualTypeClass := IBaseLibrary.TypeClass.TYPE_BOOL,
                           ActualValue := ADR(Actual),
                           Message := Message,
                           TestInstancePath := TestInstancePath,
                           AlreadyReported => AlreadyReported);
Does anyone else had this problem and know how to solve it? I'm not sure if it's something that i do wrong in my PLC library or depends entirely on the TcUnit library...
Thanks in advance for the support.

**Comment by sagatowski** [comment]
Hi @andimin182. Not enough context. Can you provide the following information:

**To Reproduce**
What steps are necessary to reproduce the behavior?

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Software versions**
Which version of the applicable software did you use? (TcUnit library, TwinCAT XAE version, Visual Studio version etc)

**Run environment**
Under what environment did you run the software? PLC? Virtual machine?

Do you have a working example of this problem that you can share?

**Reply by andimin182** [reply]
Yes, of course:

**Run ENV**
I run it both on PLC and local virtual machine.

**SW versions**
- Visual Studio 2017 v 15.9.51
- TcUnit lib 1.2.1.0

**Steps to reproduce**
I have a quiet huge FSM. I added a new functionality, i.e. a new transition from one state to another. Therefore, I wrote an additional unit test to validate this transition. I put a breackpoint just before the AssertEquals function and the test it seems correct to me as you can see from the image below. However, i have an Exception from the first assert:
![image](https://github.com/tcunit/TcUnit/assets/64167148/17c292bb-96ff-403b-820f-a3b274cdbd79)

![image](https://github.com/tcunit/TcUnit/assets/64167148/85605185-7c10-4f49-bda4-93a15de8faa1)

**Comment by sagatowski** [comment]
There is no version 1.2.1 of TcUnit. Please download of the releases, preferably [the latest one](https://github.com/tcunit/TcUnit/releases/tag/1.3.0.0).

If this doesn't work, please upload an example project that reproduces the error.

**Comment by sagatowski** [comment]
It's very hard for me to see what's happening, but is it possible that you are asserting a property?
In that case, read this: https://github.com/tcunit/TcUnit/issues/209
And this: https://tcunit.org/#/faq?id=_13-assertequalsany-on-properties-makes-my-development-environment-crash-why
